### ISSUE TITLE
Object Can't Cast to Itself

### ISSUE THREAD

**Issue by cscer** [dotnet/corefx#24280]
Message=[A]Donn.EAM.Archives.Entity.Asset cannot be cast to [B]Donn.EAM.Archives.Entity.Asset. 
Type A originates from 'Donn.EAM.Archives.Entity, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null' in the context 'Default' 
at location 'D:\MainServer\bin\Debug\netcoreapp2.0\Donn.EAM.Archives.Entity.dll'. 
Type B originates from 'Donn.EAM.Archives.Entity, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null' in the context 'Default' 
at location 'D:\MainServer\bin\Debug\netcoreapp2.0\Donn.EAM.Archives.Entity.dll'.

All assemblies and types are loaded by assembly. Load when the program is started
An error occurs when an object created by reflection is converted to a target object through the AS.
What is the difference between loading assemblies between NET Framework and net Core app,

**Comment by svick** [comment]
Can you share short code that reproduces the issue?

**Comment by karelz** [comment]
Is that a problem on .NET Core? Do you have minimal repro?

**Comment by tarekgh** [comment]
@cscer please provide a simple repro and clarify if this scenario working fine on the full .Net Framework. Thanks.

**Comment by cscer** [comment]
@karelz @tarekgh @svick 

sorry for the late reply.

yes, it's working fine on the full .Net Framework.

I tried to write a project to reproduce the problem, but did not succeed.

I think there are still a number of bugs in netcore or there are differences with the full framework, and vs2017 is not stable enough.

I've stopped migrating this project to Netstandard and reused net461.

May be the same reason as this.
https://github.com/dotnet/coreclr/issues/13969
